### PR TITLE
add value type.

### PR DESCRIPTION
**Pull request by naoki-akimoto**
ThingIFAPI#aggregateにT用のクラス引数を追加しました。レビューをお願いします。

### CODE REVIEW

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 1]
```diff
@@ -2055,6 +2055,7 @@ public String getFirmwareVersion() throws ThingIFException {
      * @param groupedQuery {@link GroupedHistoryStatesQuery} instance. timeRange in query should less than
      *                                               60 data grouping intervals.
      * @param aggregation {@link Aggregation} instance.
+     * @param fieldClass Class of aggregated result field.
```
Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: what about using `Class of {@link AggregatedResult#value}`?

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 2]
```diff
@@ -2055,6 +2055,7 @@ public String getFirmwareVersion() throws ThingIFException {
...
      * @param <T> Type of aggregated result field.
```
Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: what about fix `<T>`
```java
@param <T> Type of aggregated result field.
```
to
```java
@param <T> Type of {@link AggregatedResult#value}.

```

**Comment by naoki-akimoto** [comment]
@yongpingchen fixed. Please confirm.

**Comment by shinsuke-nara** [comment]
@naoki-akimoto LGTM

@yongpingchen Should we talk about this with @satoshikumano san?

**Comment by yongpingchen** [comment]
> Should we talk about this with @satoshikumano san?

Sure.

**Comment by yongpingchen** [comment]
@naoki-akimoto , after discussed with @satoshikumano and @shinsuke-nara ,  it is difficult to cast type without class instance as parameter, so we  will adjust the design for the following APIs: 
1. `<T extends Number, S extends TargetState> List<AggregatedResult<T, S>> ThingIFAPI#aggregate(GroupedHistoryStatesQuery, Aggregation)` -> `<T extends Number, S extends TargetState> List<AggregatedResult<T, S>> ThingIFAPI#aggregate(GroupedHistoryStatesQuery, Aggregation, Class<S>, Class<T>)`
1. `<S extends TargetState> Pair<List<HistoryState<S>>, String> ThingIFAPI#query(HistoryStateQuery,)` -> `<S extends TargetState> Pair<List<HistoryState<S>>, String> ThingIFAPI#query(HistoryStateQuery, Class<S>)`
1. `<S extends TargetState> Pair<List<HistoryState<S>>, String> ThingIFAPI#query(HistoryStateQuery,)` -> `<S extends TargetState> Pair<List<HistoryState<S>>, String> ThingIFAPI#query(GroupedHistoryStatesQuery, Class<S>)` 

Please continue the implementation of current design, I will adjust the new API design in another PR.

**Pull request merged**